contracts/vending-minter/src/validation.rs

```diff
@@ -0,0 +1,41 @@
+use crate::ContractError;
+use crate::ContractError::CheckedMultiplyFractionError;
+use cosmwasm_std::Uint128;
+
+pub fn get_three_percent_of_tokens(num_tokens: u32) -> Result<Uint128, ContractError> {
+    let three_percent = (Uint128::new(3), Uint128::new(100));
+    let three_percent_tokens = Uint128::from(num_tokens)
+        .checked_mul_ceil(three_percent)
+        .map_err(|_| CheckedMultiplyFractionError {})?;
+    Ok(three_percent_tokens)
+}
+pub fn get_max_mintable_tokens(
+    per_address_limit: u32,
+    num_tokens: u32,
+    max_per_address_limit: u32,
+) -> Result<u32, ContractError> {
+    if per_address_limit > max_per_address_limit {
+        return Ok(max_per_address_limit);
```

Comment:
This is basically just a check to see if per_address_limit is over 50, since max_per_address_limit is harded to 50. This however is already being tested here: 

```rs
    // 3% of 5000 is 150, but we will get blocked by the max per address limit flag
    let per_address_limit_msg = ExecuteMsg::UpdatePerAddressLimit {
        per_address_limit: 51,
    };
    let res = router.execute_contract(
        creator.clone(),
        minter_addr.clone(),
        &per_address_limit_msg,
        &[],
    );
    assert!(res.is_err());
    ```

Reply:
Updated test name to be more clear: `fn check_per_address_limit_above_max_per_address_limit() {`

Reply:
This function is only used to display in the error msg back to user what the max allowed mintable tokens should be. 

This error is being returned in the unit test `check_invalid_update_per_address_limit`. It sets the per address limit to 110, but what is returned back in the error message is max_per_address_limit which is 50. 
```rs

    let update_msg = ExecuteMsg::UpdatePerAddressLimit {
        per_address_limit: 110,
    };
    let err = router
        .execute_contract(creator, minter_addr, &update_msg, &[])
        .unwrap_err();
    assert_eq!(
        err.source().unwrap().to_string(),
        ContractError::InvalidPerAddressLimit {
            max: max_per_address_limit,
            min: 1,
            got: 110,
        }
        .to_string()
    );
    ```

Reply:
I have updated the name of the function to be more explicit, I also moved it out of validation and into contracts.rs: 
`display_max_mintable_tokens`